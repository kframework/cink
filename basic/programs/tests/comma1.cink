int main () {
   int i = 10, b = 20, c= 30;
   i = b, c;
   cout << i;

   i = (b, c);
   cout << i;
}
