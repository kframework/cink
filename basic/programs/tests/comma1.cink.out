2030<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda (  @ int i ; (i = 10) ; int b ; (b = 20)
           ; int c ; (c = 30) ; ((i = b) , c) ; (cout << i) ; (i = (b , c)) ; 
          (cout << i) ; )
        #symScalLoc(2) |-> 30
        #symScalLoc(3) |-> 20
        #symScalLoc(4) |-> 30
    </store>
</T>
